### ISSUE TITLE
Question on Equation and Implementation

### ISSUE THREAD

**Issue by Rongjiehuang** [cientgu/VQ-Diffusion#15]
Hi author, thanks for sharing your inspiring work!

I found that some codes could be hard to follow. For example, in [line 182](https://github.com/cientgu/VQ-Diffusion/blob/Improved_VQ-Diffusion/image_synthesis/modeling/transformers/diffusion_transformer.py#L182), what is the goal of this _log_add_exp_ function, and how this calculation of _log_probs_ corresponds to equations in the paper? It seems that you implemented VQ diffusion all in log scale, which is different from the original DDPM to a large extent. Could you explain Why?

Thanks. Looking forward to your reply.

**Comment by cientgu** [comment]
It's inspired by multinomail_diffusion(https://github.com/ehoogeboom/multinomial_diffusion). log scale could make the computation more stable. You may refer to their code and paper for more details.
